packages/wallet-sdk/src/features/sub-accounts/createSubAccountSigner.ts

```diff
@@ -0,0 +1,109 @@
+import { Hex, http, SignableMessage, TypedDataDefinition } from 'viem';
+import {
+  createPaymasterClient,
+  toCoinbaseSmartAccount,
+  ToCoinbaseSmartAccountReturnType,
+} from 'viem/account-abstraction';
+import { baseSepolia } from 'viem/chains';
+
+import { SubAccount } from './state.js';
+import { toViemAccount } from './toViemAccount.js';
+import { AddAddressResponse } from './types.js';
+import { RequestArguments } from ':core/provider/interface.js';
+import { getBundlerClient, getClient } from ':features/clients/utils.js';
+
+async function toSmartAccount({
+  subAccount,
+}: {
+  subAccount: AddAddressResponse;
+}): Promise<ToCoinbaseSmartAccountReturnType> {
+  const { getSigner, getAddress, type } = SubAccount.getState();
+  if (!getSigner) {
+    throw new Error('get signer not found');
+  }
+
+  const sign = await getSigner();
+  if (!sign) {
+    throw new Error('signer not found');
+  }
+
+  const signerAddress = await getAddress();
+  const account = toViemAccount({
+    type,
+    sign,
+    address: signerAddress,
+  });
+
+  const client = getClient(subAccount.chainId ?? baseSepolia.id);
```

Comment:
can we add a comment to remind us to change this to base when we release this feature

Reply:
good callout